src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs

```diff
@@ -621,5 +637,127 @@ public static void ReturnState(ref ulong[] state)
                 state = Array.Empty<ulong>();
             }
         }
+
+        public static void KeccakF1600Avx512F(Span<ulong> state)
+        {
+            {
+                // Redundant statement that removes all the in loop bounds checks
+                _ = state[24];
+            }
+
+            // Can straight load and over-read for start elements
+            Vector512<ulong> c0 = Unsafe.As<ulong, Vector512<ulong>>(ref MemoryMarshal.GetReference(state));
+            Vector512<ulong> c1 = Unsafe.As<ulong, Vector512<ulong>>(ref Unsafe.Add(ref MemoryMarshal.GetReference(state), 5));
+            Vector512<ulong> c2 = Unsafe.As<ulong, Vector512<ulong>>(ref Unsafe.Add(ref MemoryMarshal.GetReference(state), 10));
+            Vector512<ulong> c3 = Unsafe.As<ulong, Vector512<ulong>>(ref Unsafe.Add(ref MemoryMarshal.GetReference(state), 15));
+            // Can't over-read for the last elements (8 items in vector 5 to be remaining)
+            // so read them directly as ulongs
+            Vector512<ulong> c4 = Vector512.Create(state[20], state[21], state[22], state[23], state[24], 0UL, 0UL, 0UL);
```

Comment:
Can you use Vector256 for first 4 bytes?

Reply:
Good call; asm looks better for load and save for the last Vector